core/src/main/java/org/apache/iceberg/MetricsUtil.java

```diff
@@ -56,4 +64,123 @@ public static MetricsModes.MetricsMode metricsMode(
     String columnName = inputSchema.findColumnName(fieldId);
     return metricsConfig.columnMode(columnName);
   }
+
+  /**
+   * Return a readable metrics map
+   *
+   * @param schema schema of original data table
+   * @param namesById pre-computed map of all column ids in schema to readable name, see {@link
+   *     org.apache.iceberg.types.TypeUtil#indexNameById(Types.StructType)}
+   * @param contentFile content file with metrics
+   * @return map of readable column name to column metric, of which the bounds are made readable
+   */
+  public static Map<String, StructLike> readableMetricsMap(
+      Schema schema, Map<Integer, String> namesById, ContentFile<?> contentFile) {
+    Map<String, StructLike> metricsStruct = Maps.newHashMapWithExpectedSize(namesById.size());
+
+    Map<Integer, Long> columnSizes = contentFile.columnSizes();
+    Map<Integer, Long> valueCounts = contentFile.valueCounts();
+    Map<Integer, Long> nullValueCounts = contentFile.nullValueCounts();
+    Map<Integer, Long> nanValueCounts = contentFile.nanValueCounts();
+    Map<Integer, ByteBuffer> lowerBounds = contentFile.lowerBounds();
+    Map<Integer, ByteBuffer> upperBounds = contentFile.upperBounds();
+
+    for (int id : namesById.keySet()) {
+      Types.NestedField field = schema.findField(id);
+      if (field.type().isPrimitiveType()) {
+        // Iceberg stores metrics only for primitive types
+        String colName = namesById.get(id);
+        ReadableMetricsStruct struct =
+            new ReadableMetricsStruct(
+                columnSizes == null ? null : columnSizes.get(id),
+                valueCounts == null ? null : valueCounts.get(id),
+                nullValueCounts == null ? null : nullValueCounts.get(id),
+                nanValueCounts == null ? null : nanValueCounts.get(id),
+                lowerBounds == null ? null : convertToReadable(field, lowerBounds.get(id)),
+                upperBounds == null ? null : convertToReadable(field, upperBounds.get(id)));
+        metricsStruct.put(colName, struct);
+      }
+    }
+    return metricsStruct;
```

Comment:
nit: What about an empty line before the return statement to separate the for loop above?